airbyte-webapp/src/components/connection/ConnectionForm/Section.tsx

```diff
@@ -1,26 +1,61 @@
+import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import classNames from "classnames";
+import React, { useState } from "react";
 
+import { Button } from "components/ui/Button";
 import { Card } from "components/ui/Card";
 import { Heading } from "components/ui/Heading";
 
 import styles from "./Section.module.scss";
 
-interface SectionProps {
+export interface SectionProps {
   title?: React.ReactNode;
   flush?: boolean;
   className?: string;
+  collapsible?: boolean;
+  collapsedPreviewInfo?: React.ReactNode;
+  collapsedInitially?: boolean;
+  testId?: string;
 }
 
-export const Section: React.FC<React.PropsWithChildren<SectionProps>> = ({ title, flush, children, className }) => {
+export const Section: React.FC<React.PropsWithChildren<SectionProps>> = ({
+  title,
+  flush,
+  children,
+  className,
+  collapsible = false,
+  collapsedPreviewInfo,
+  collapsedInitially = collapsible,
```

Comment:
I'm not sure this is good behavior.
Saying something is collapsible should not automatically collapse it.
However based on the logic below, I think it might be safe to have `collapsible = collapsedInitially` instead.
I'd almost rather not do that at all though, and emit a warning if `collapsedInitially` is true and `collapsible` is false.

Reply:
Agree, and I guess the last option would be the best choice. I've added a warning for the case when `collapsedInitially && !collapsible`